framework-crates/objc2-metal/src/device.rs

```diff
@@ -1,16 +1,24 @@
-// use crate::MTLDevice;
-//
-// pub trait MTLDeviceExtension {
-//     pub fn system_default() -> Option<Self> {
-//         MTLCreateSystemDefaultDevice()
+use objc2::{rc::Id, runtime::ProtocolObject};
+use objc2_foundation::NSArray;
+
+use crate::MTLDevice;
+
+// pub trait MTLDeviceExt: MTLDevice {
+//     fn system_default() -> Option<Id<ProtocolObject<dyn MTLDevice>>> {
+//         unsafe { Id::from_raw(crate::MTLCreateSystemDefaultDevice()) }
 //     }
-//
-//     pub fn all() -> Retained<NSArray<Self>> {
-//         #[cfg(target_os = "macos")]
-//         MTLCopyAllDevices()
-//         #[cfg(not(target_os = "macos"))]
-//         NSArray::from(MTLCreateSystemDefaultDevice())
+
+//     fn all() -> Id<NSArray<ProtocolObject<dyn MTLDevice>>> {
+//         unsafe { Id::new_nonnull(crate::MTLCopyAllDevices()) }
 //     }
 // }
-//
-// impl<P: MTLDevice> MTLDeviceExtension for P {}
+
+// impl<P: MTLDevice> MTLDeviceExt for P {}
+
+pub fn mtl_device_system_default() -> Option<Id<ProtocolObject<dyn MTLDevice>>> {
```

Comment:
I would rather have these be something like:
```rust
pub fn MTLCreateSystemDefaultDevice() -> Option<Retained<ProtocolObject<dyn MTLDevice>>> {
    extern "C" {
        pub fn MTLCreateSystemDefaultDevice() -> *mut ProtocolObject<dyn MTLDevice>;
    }
    unsafe { Retained::from_raw(MTLCreateSystemDefaultDevice()) }
}
```

That is, change the signature of `MTLCreateSystemDefaultDevice` to do the `Retained::from_raw` step internally.

Reply:
That would be lovely! I take it from this that we should keep the name to match Objective-C with CamelCase too?

Let me know when you can set that up in the generator, which would save us the hassle of reimplementing. But if you don't get to that in time, landing this now should make it easier to notice that there's some files that can be removed in the future.

Reply:
Hmm, I've been (and will be) away for various summer holiday activities in recent times, so I'm a bit out of the loop myself. I think it would be fairly easy to implement generally, but I probably won't have time to do it before August.

I'm fine with pulling this change _now_, but in any case, I'm probably not going to cut a new release before it's implemented generally, so you could also just wait for that.

Reply:
I don't absolutely need these functions right now (would be nice to start using them in the example in https://github.com/Traverse-Research/gpu-allocator/pull/225).

Shall I leave this PR open then to not forget to remove this custom file when you've resolved #639? Thanks in advance!